src/spanningtrees/planar_maximally_filtered_graph.jl

```diff
@@ -0,0 +1,74 @@
+"""
+	planar_maximally_filtered_graph(g)
+
+Compete the Planar Maximally Filtered Graph (PMFG) of weighted graph `g`.
+
+### Examples
+```
+using Graphs, SimpleWeightedGraphs
+N = 20
+M = Symmetric(randn(N, N))
+g = SimpleWeightedGraph(M) 
+p_g = planar_maximally_filtered_graph(g)
+```
+
+### References 
+- Tuminello et al. 2005
+"""
+function planar_maximally_filtered_graph end
+@traitfn function planar_maximally_filtered_graph(
+    g::AG::(!IsDirected), distmx::AbstractMatrix{T}=weights(g); minimize=true
+) where {T<:Real,U,AG<:AbstractGraph{U}}
+    #check graph is not directed
+    if is_directed(g)
+        error("PMFG only supports non-directed graphs")
+    end
+
+    #construct a list of edge weights
+    weights = Vector{T}()
+    sizehint!(weights, ne(g))
+    edge_list = collect(edges(g))
+    for e in edge_list
+        push!(weights, distmx[src(e), dst(e)])
+    end
+    #sort the set of edges by weight
+    #we try to maximally filter the graph and assume that weights
+    #represent distances. Thus we want to add edges with the 
+    #smallest distances first. Given that we pop the edge list, 
+    #we want the smallest weight edges at the end of the edge list
+    #after sorting, which means reversing the usual direction of
+    #the sort.
+    edge_list .= edge_list[sortperm(weights; rev=minimize)]
+
+    #construct an initial graph
+    test_graph = SimpleGraph(nv(g))
+
+    #go through the edge list 
+    while !isempty(edge_list)
+        e = pop!(edge_list) #get most weighted edge
```

Comment:
```suggestion
   for e in edge_list
```
might be easier - although when you used `pop!` you always popped the last element of the vector, so we would need to reverse the order of the sort.